codalab/server/bundle_manager.py

```diff
@@ -977,25 +1039,19 @@ def _get_staged_bundles_to_run(self, workers, user_info_cache):
                 # temporarily, we filter out those bundles so that they won't be dispatched to run on workers.
                 if len(matched_workers) == 0:
                     if bundle.uuid not in self._bundles_without_matched_workers:
-                        self._model.update_bundle(
-                            bundle,
-                            {
-                                'metadata': {
-                                    'staged_status': 'Bundle is requested to run on a worker {} which has '
-                                    'not been connected to the CodaLab server yet.'.format(
-                                        bundle.metadata.request_queue
-                                    )
-                                }
-                            },
+                        staged_status = 'Bundle is requested to run on a worker {} which has not been connected to the CodaLab server yet.'.format(
```

Comment:
I feel like this logic shouldn't be here; if queue is handled as part of resources, this should be in the part of the same part of the code that reports when there aren't enough GPUs/CPUs, etc.